File: core/src/nl/t64/game/rpg/screens/menu/ListenerKeyConfirm.java
package nl.t64.game.rpg.screens.menu;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.InputListener;
import nl.t64.game.rpg.constants.Constant;


class ListenerKeyConfirm extends InputListener {

    private final ListenerInput listenerInput;

    ListenerKeyConfirm(Runnable selectItemFunction) {
        this.listenerInput = new ListenerInput(selectItemFunction);
    }

    @Override
    public boolean keyDown(InputEvent event, int keycode) {
        switch (keycode) {
            case Constant.KEYCODE_BOTTOM,
                    Input.Keys.ENTER -> listenerInput.inputConfirm();
        }
        return true;
    }

}

File: core/src/nl/t64/game/rpg/screens/menu/ListenerKeyConfirm.kt
package nl.t64.game.rpg.screens.menu

import com.badlogic.gdx.Input
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.InputListener
import nl.t64.game.rpg.constants.Constant


internal class ListenerKeyConfirm(selectItemFunction: () -> Unit) : InputListener() {

    private val listenerInput: ListenerInput = ListenerInput(selectItemFunction)

    override fun keyDown(event: InputEvent, keycode: Int): Boolean {
        when (keycode) {
            Constant.KEYCODE_BOTTOM, Input.Keys.ENTER -> listenerInput.inputConfirm()
        }
        return true
    }

}
